storage/database/sharded_database_test.go

```diff
@@ -0,0 +1,270 @@
+package database
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"sort"
+	"sync"
+	"testing"
+
+	"github.com/klaytn/klaytn/common"
+	"github.com/stretchr/testify/assert"
+)
+
+var ShardedDBConfig = []*DBConfig{
+	{DBType: LevelDB, SingleDB: false, NumStateTrieShards: 4, ParallelDBWrite: true},
+	{DBType: LevelDB, SingleDB: false, NumStateTrieShards: 8, ParallelDBWrite: true},
+	{DBType: LevelDB, SingleDB: false, NumStateTrieShards: 16, ParallelDBWrite: true},
+}
+
+func createEntries(entryNum int) []entry {
+	entries := make([]entry, entryNum)
+	for i := 0; i < entryNum; i++ {
+		entries[i] = entry{common.MakeRandomBytes(256), common.MakeRandomBytes(600)}
+	}
+	return entries
+}
+
+// testIterator tests if given iterator iterates all entries
+func testIterator(t *testing.T, checkOrder bool, entriesFromIterator func(db shardedDB, entryNum int) []entry) {
+	entryNum := 500
+	entries := createEntries(entryNum)
+	dbs := make([]shardedDB, len(ShardedDBConfig))
+
+	// create DB and write data for testing
+	for i, config := range ShardedDBConfig {
+		config.Dir, _ = ioutil.TempDir(os.TempDir(), "test-shardedDB-iterator")
+		defer func(dir string) {
+			if err := os.RemoveAll(dir); err != nil {
+				t.Fatalf("fail to delete file %v", err)
+			}
+		}(config.Dir)
+
+		// create sharded DB
+		db, err := newShardedDB(config, 0, config.NumStateTrieShards)
+		dbs[i] = *db
+		if err != nil {
+			t.Log("Error occured while creating DB")
+			t.FailNow()
+		}
```

Comment:
Although it is a testcase, it is better to check the error first.
```suggestion
		if err != nil {
			t.Log("Error occured while creating DB")
			t.FailNow()
		}
		dbs[i] = *db
```

Reply:
The logic is erased. Thank you for the comment.